main.go

```diff
@@ -16,13 +16,16 @@ var content embed.FS // object representing the embedded directory
 
 var log *slog.Logger
 
-func main() {
+func init() {
```

Comment:
You answered this already but my question was what is the difference between `main` and `init`

Reply:
`init` can come with weird side effects.

consider go's order of imports
![image](https://github.com/user-attachments/assets/a620d297-b46c-4b7b-9eaa-5b2d5304b76e)


What I'm doing currently is defining a global `var log *slog.Logger`, a pointer to our logging library, and then writing to the value of that global in `init()`. It's a little more straightforward, and less prone to confusion or "shadowing" (overwriting) the variable to instead use a pattern of returning a new object in main.

I do that in the api like
```go
log, err := logs.New()
```